pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -0,0 +1,133 @@
+package onbuild
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+	"github.com/openshift/source-to-image/pkg/build"
+	"github.com/openshift/source-to-image/pkg/docker"
+	"github.com/openshift/source-to-image/pkg/git"
+	"github.com/openshift/source-to-image/pkg/tar"
+	"github.com/openshift/source-to-image/pkg/util"
+)
+
+// OnBuild strategy executes the simple Docker build in case the image does not
+// support STI scripts but has ONBUILD instructions recorded.
+type OnBuild struct {
+	docker  docker.Docker
+	git     git.Git
+	fs      util.FileSystem
+	tar     tar.Tar
+	source  build.Downloader
+	garbage build.Cleaner
+}
+
+// NewOnBuild returns a new instance of OnBuild builder
+func NewOnBuild(request *api.Request) (*OnBuild, error) {
+	dockerHandler, err := docker.NewDocker(request.DockerSocket)
+	if err != nil {
+		return nil, err
+	}
+	b := &OnBuild{
+		docker: dockerHandler,
+		git:    git.NewGit(),
+		fs:     util.NewFileSystem(),
+		tar:    tar.NewTar(),
+	}
+	b.source = &git.Clone{b.git, b.fs}
+	b.garbage = &build.DefaultCleaner{b.fs, b.docker}
+	return b, nil
+}
+
+// SourceTar produces a tar archive containing application source and stream it
+func (b *OnBuild) SourceTar(request *api.Request) (io.ReadCloser, error) {
+	uploadDir := filepath.Join(request.WorkingDir, "upload", "src")
+	tarFileName, err := b.tar.CreateTarFile(request.WorkingDir, uploadDir)
+	if err != nil {
+		return nil, err
+	}
+	return b.fs.Open(tarFileName)
+}
+
+// Build executes the ONBUILD kind of build
+func (b *OnBuild) Build(request *api.Request) (*api.Result, error) {
+	glog.V(2).Info("Preparing the source code for build")
+	if err := b.Prepare(request); err != nil {
```

Comment:
What is the reason for not doing that? I don't see any reason why we need to hide this method. It can be useful for other builders as well (I want a builder that use STI as Preparer)... Make methods private cause nobody can consume them and it also makes the code less extensible.